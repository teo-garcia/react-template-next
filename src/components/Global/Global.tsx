import { GlobalProps } from "../../types/client";

function Global(props: GlobalProps) {
  const { children } = props;
  return <>{children}</>;
}

export { Global };
