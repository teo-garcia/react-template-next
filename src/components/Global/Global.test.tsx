import { render, screen } from '@testing-library/react';
import { Global } from './Global';

describe('<Global /> tests', function () {
  test('Should match with the component snapshot', function () {
    const { asFragment } = render(<Global>Hello World</Global>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should render the provided children prop', function () {
    render(<Global>Hello World</Global>);
    expect(screen.getByText(/Hello World/i)).toBeInTheDocument();
  });
});
