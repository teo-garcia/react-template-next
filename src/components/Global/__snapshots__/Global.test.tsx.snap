// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Global /> tests Should match with the component snapshot 1`] = `
<DocumentFragment>
  Hello World
</DocumentFragment>
`;
